feat: add vimtex integration

Vinzent03 opened the pull request:
This pr adds an integration for the [vimtex](https://github.com/lervag/vimtex) plugin. This adds quite more keymaps than the other integrations, and it's more complicated to simulate the original key press, because it does not provide a lua interface.
So I'm curious what you think about this integration.

For me, a nice addition with this integration is a better description in which-key, because it now shows the actual action and not just `vimtex-]])`.

Vinzent03 committed:
docs: auto generate vimdoc

Vinzent03 committed:
feat: add vimtex integration

Vinzent03 committed:
refactor: rename to begin_start

mawkler commented on lua/demicolon/integrations/vimtex.lua:
```diff
@@ -0,0 +1,103 @@
+function M.jump(options)
+  return function()
+    require('demicolon.jump').repeatably_do(function(opts)
+      if vim.fn.exists(':VimtexCompile') ~= 2 then
```
Perhaps we can do this check just once from the `Filetype tex` autocmd callback in instead, and not attach any keymaps if vimtex is not found?

Vinzent03 replied:
I wasn't use if we encounter any race conditions here, because vimtex loads itself for the tex filetype, but in my testing it always worked, so I guess it is fine and better this way.

mawkler commented on lua/demicolon/integrations/vimtex.lua:
```diff
@@ -0,0 +1,103 @@
+        return
+      end
+
+      local direction = (opts.forward == nil or opts.forward) and 'next' or 'prev'
+
+      local vimtex_key = opts.vimtex_key
+      local vimtex_mapping = ""
+
+      if direction == "prev" then
+        vimtex_mapping = "[" .. vimtex_key
+      else
+        vimtex_mapping = "]" .. vimtex_key
+      end
```
```suggestion
      local direction = (opts.forward == nil or opts.forward) and ']' or '['
      local vimtex_mapping = direction .. opts.vimtex_key
```

mawkler replied:
Here's a little shorter implementation

mawkler committed:
docs: mention lazy loading on tex filetype

mawkler reviewed (changes requested):
Thank you for this PR. And thank you for providing great documentation!

Could you also add these two things to the README.md please?

1. Update the configuration table in the [default options](https://github.com/mawkler/demicolon.nvim?tab=readme-ov-file#configuration) so that it includes the vimtex options
2. Add a Vimtex section to [Usage](https://github.com/mawkler/demicolon.nvim?tab=readme-ov-file#usage) with a table with mappings and explanations

Could you also make sure that you use `'`s instead of `"` for your strings for formatting consistency?

Vinzent03 committed:
refactor: use simpler code

Co-authored-by: Melker Ulander <melker.ulander@pm.me>

Vinzent03 committed:
style: use single quotes

Vinzent03 committed:
fix: move vimtex check to tex autocmd

Vinzent03 committed:
fix: use correct mapping for comment end

Vinzent03 committed:
docs: update readme

Vinzent03 committed:
fix: only map in normal mode

Vinzent03 commented:
Great, all your requested changes should be implemented.
Additionally, I've fixed the keymap for comment end and changed the mode to normal only, because I couldn't get it to work in operator pending mode.

mawkler commented:
Awesome!

Yeah that's odd that the operator pending-mapping doesn't work, I tried it as well. Perhaps we should keep the visual mode mapping as well (`'x'`)? That one seems to work

Vinzent03 committed:
fix: map in x mode as well

Vinzent03 commented:
Oh yeah you are right. I've added it back in. I've also added a note to the readme.

mawkler committed:
docs: expand on omission of operator-pending mode

mawkler commented:
Great work! Thank you very much for your contribution!
